internal/rpc/jsonrpc/methods.go

```diff
@@ -2049,6 +2050,113 @@ func (s *Server) getTransaction(ctx context.Context, icmd interface{}) (interfac
+	cmd := icmd.(*types.GetTxOutCmd)
+	w, ok := s.walletLoader.LoadedWallet()
+	if !ok {
+		return nil, errUnloadedWallet
+	}
+
+	// Attempt RPC passthrough if connected to DCRD.
+	n, err := w.NetworkBackend()
+	if err != nil {
+		return nil, err
+	}
+	if rpc, ok := n.(*dcrd.RPC); ok {
+		var resp json.RawMessage
+		err := rpc.Call(ctx, "gettxout", &resp, cmd.Txid, cmd.Vout, cmd.Tree, cmd.IncludeMempool)
+		return resp, err
+	}
+
+	txHash, err := chainhash.NewHashFromStr(cmd.Txid)
+	if err != nil {
+		return nil, rpcError(dcrjson.ErrRPCDecodeHexString, err)
+	}
+
+	if cmd.Tree != wire.TxTreeRegular && cmd.Tree != wire.TxTreeStake {
+		return nil, rpcErrorf(dcrjson.ErrRPCInvalidParameter, "Tx tree must be regular or stake")
+	}
+
+	// Check if the transaction is known to wallet.
+	walletTx, err := wallet.UnstableAPI(w).TxDetails(ctx, txHash)
+	if err != nil && !errors.Is(err, errors.NotExist) {
+		return nil, err
+	}
+	if walletTx == nil {
+		return nil, nil // Tx not found in wallet.
+	}
+	if len(walletTx.MsgTx.TxOut) <= int(cmd.Vout) {
+		return nil, rpcErrorf(dcrjson.ErrRPCInvalidTxVout, "invalid vout %d", cmd.Vout)
+	}
+
+	tx := &walletTx.MsgTx
+	txTree := wire.TxTreeRegular
+	if stake.DetermineTxType(tx, true) != stake.TxTypeRegular {
+		txTree = wire.TxTreeStake
+	}
+	if txTree != cmd.Tree {
+		// Not an error because it is technically possible (though extremely unlikely)
+		// that the required tx (same hash, different tree) exists on the blockchain.
+		return nil, nil
+	}
+
+	// Attempt to read the unspent txout info from wallet.
+	outpoint := wire.OutPoint{Hash: *txHash, Index: cmd.Vout, Tree: cmd.Tree}
+	walletUnspent, err := w.UnspentOutput(ctx, outpoint, *cmd.IncludeMempool)
+	if err != nil && !errors.Is(err, errors.NotExist) {
+		return nil, err
+	}
+	if walletUnspent == nil {
+		return nil, nil // output is spent
+	}
+
+	txout := tx.TxOut[cmd.Vout]
+	pkScript := txout.PkScript
+	scriptVersion := txout.Version
+
+	// Disassemble script into single line printable format.  The
+	// disassembled string will contain [error] inline if the script
+	// doesn't fully parse, so ignore the error here.
+	disbuf, _ := txscript.DisasmString(pkScript)
+
+	// Get further info about the script.  Ignore the error here since an
+	// error means the script couldn't parse and there is no additional
+	// information about it anyways.
+	scriptClass, addrs, reqSigs, _ := txscript.ExtractPkScriptAddrs(
+		scriptVersion, pkScript, s.activeNet, true) // Yes treasury
+	addresses := make([]string, len(addrs))
+	for i, addr := range addrs {
+		addresses[i] = addr.Address()
+	}
+
+	bestHash, bestHeight := w.MainChainTip(ctx)
+	var confirmations int64
+	if walletTx.Block.Height != -1 {
+		confirmations = int64(1 + bestHeight - walletTx.Block.Height)
+	}
+
+	return &dcrdtypes.GetTxOutResult{
+		BestBlock:     bestHash.String(),
+		Confirmations: confirmations,
+		Value:         dcrutil.Amount(txout.Value).ToUnit(dcrutil.AmountCoin),
```

Comment:
Although this is verbatim from dcrd's rpcserver.go, it's a bit odd that it uses `ToUnit(dcrutil.AmountCoin)` instead of `ToCoin()`.  Maybe clearer to shorten this?